topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClient.java

```diff
@@ -0,0 +1,158 @@
+     * Send a GET request to {@code user_tasks} endpoint.
+     * This can be used to check the asynchronous task execution result.
+     * 
+     * @param userTaskIds List of user task ids.
+     * @return User tasks response.
+     */
+    UserTasksResponse userTasks(List<String> userTaskIds);
+
+    /**
+     * Get the error message from HTTP response.
+     * 
+     * @param response The HTTP response.
+     * @return The error message.
+     */
+    Optional<String> errorMessage(HttpResponse<String> response);
+    
+    /**
+     * Topic names grouped by replication factor value.
+     * 
+     * @param topicByReplicationFactor Topic names grouped by replication factor value.
+     */
+    record ReplicationFactor(@JsonProperty("topic_by_replication_factor") Map<Integer, String> topicByReplicationFactor) { }
+
+    /**
+     * Replication factor changes.
+     * 
+     * @param replicationFactor Replication factor value.
+     */
+    record ReplicationFactorChanges(@JsonProperty("replication_factor") ReplicationFactor replicationFactor) { }
+    
+    /**
+     * The user task.
+     * 
+     * @param status Status.
+     * @param clientIdentity Client identity.
+     * @param requestURL Request URL.
+     * @param userTaskId User task id.
+     * @param startMs Start time in ms.
+     */
+    record UserTask(
+        @JsonProperty("Status") String status,
+        @JsonProperty("ClientIdentity") String clientIdentity,
+        @JsonProperty("RequestURL") String requestURL,
+        @JsonProperty("UserTaskId") String userTaskId,
+        @JsonProperty("StartMs") String startMs
+    ) { }
+
+    /**
+     * The user tasks response.
+     * 
+     * @param userTasks User task list.
+     * @param version Version.
+     */
+    record UserTasksResponse(List<UserTask> userTasks, int version) { }
+
+    /**
+     * The error response.
+     * 
+     * @param stackTrace Stack trace.
+     * @param errorMessage Error message.
+     * @param version Version.
+     */
+    record ErrorResponse(String stackTrace, String errorMessage, int version) { }
+
+    /**
+     * Task states.
+     */
+    enum TaskState {
+        /**
+         * The task has been accepted and waiting for execution.
+         */
+        ACTIVE("Active"),
+
+        /**
+         * The task is being executed.
+         */
+        IN_EXECUTION("InExecution"),
+
+        /**
+         * The task has been completed.
+         */
+        COMPLETED("Completed"),
+
+        /**
+         * The task has been completed with errors.
+         */
+        COMPLETED_WITH_ERROR("CompletedWithError");
+
+        private static final List<TaskState> CACHED_VALUES = List.of(values());
+        private final String value;
+        TaskState(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+
+        /**
+         * Use this instead of values() to avoid creating a new array each time.
```

Comment:
Is that actually a problem?

Reply:
I think that unless you know it's a perf problem it might be better to leave this out and have more readable code at the call site.